doc/life_of_a_language_feature.md

```diff
@@ -1,116 +1,141 @@
-# Dart language evolution process
+# Lifecycle of a Dart Language Feature
 
-## User issue or feature request
+This document describes the process from creation to the shipping and release of
+a language feature. This document serves as a walkthrough for what's needed to
+build a language feature.
 
-Features and changes arise from perceived user issues or feature requests.
-Feature requests should be filed in the [language
-repo](https://github.com/dart-lang/language/issues/new?labels=request), and are
-labelled [`request`](https://github.com/dart-lang/language/labels/request).
-We may close issues that we believe that we will not
-address for whatever reason, or we may keep issues open indefinitely if we
-believe they are something that we may wish to address in the future. Feature
-request issues are primarily for documentation of the user issue to be solved.
+1. [User Issues and Feature Requests](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#user-issues-and-feature-requests)
+2. [Design, Feedback, and Iteration](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#design-feedback-and-iteration)
+3. [Acceptance](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#acceptance)
+4. [Team Communication](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#team-communication) (eg. kick-off meetings) 
+5. [Implementation and Testing](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#implementation-and-testing)
+6. [Migrations](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#migrations)
+7. [Shipping](https://github.com/dart-lang/language/blob/main/doc/life_of_a_language_feature.md#shipping)
```

Comment:
Once I learned that GitHub will auto-generate a live TOC for you:

<img width="83" alt="Screenshot 2024-11-01 at 5 40 41 PM" src="https://github.com/user-attachments/assets/e97bd277-7774-44a3-b0d2-b72ba3fd61d5">

I stopped hand-maintaining them. I won't tell you how long it took me to discover this feature.

Reply:
Ah yeaah, I knew about that feature, but purposefully made a TOC. If we prefer, we can remove it, I just really like showing all the sections up front at the start of the document.

Reply:
Keeping SGTM if you like it.